### PR TITLE
Changed 'int' type to 'integer' in to_numeric docstring

### LINKED ISSUE #35760: DOC: downcast argument for pd.to_numeric should be integer, not int
#### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.to_numeric.html?highlight=to_numeric#pandas.to_numeric

#### Documentation problem

The documentation declares that the `downcast` argument of `pd.to_numeric` can use the value `int`. However, only `integer` is working in the real code:

```
In [8]: pd.to_numeric(["42"], downcast="int", errors="coerce")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-3b178e77dcd5> in <module>
----> 1 pd.to_numeric(["42"], downcast="int", errors="coerce")

/usr/lib/python3.8/site-packages/pandas/core/tools/numeric.py in to_numeric(arg, errors, downcast)
    110     """
    111     if downcast not in (None, "integer", "signed", "unsigned", "float"):
--> 112         raise ValueError("invalid downcasting method provided")
    113 
    114     if errors not in ("ignore", "raise", "coerce"):

ValueError: invalid downcasting method provided

In [9]: pd.to_numeric(["42"], downcast="integer", errors="coerce")
Out[9]: array([42], dtype=int8)
```

#### Suggested fix for documentation

Replace `int` with `integer`

### PR DESCRIPTION
**Pull request by edwardkong**
- [x] closes #35760 
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by dsaxton** [comment]
Thanks @edwardkong. The docstring for SparseArray also seems to've been changed in error by the same PR, can you fix that here also?

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
Thanks @edwardkong for the PR. lgtm ex @dsaxton comment

can you also revert the changes to pandas/core/arrays/sparse/array.py made in #33093

**Comment by simonjayhawkins** [comment]
i'll add this to the 1.1.1 milestone as effectively a regression in the docs. feel free to change if disagree

**Review by jreback — APPROVED** [review]
lgtm

**Comment by simonjayhawkins** [comment]
> can you also revert the changes to pandas/core/arrays/sparse/array.py made in #33093

Thanks @edwardkong Since I want to make sure this is fixed for 1.1.1 I pushed this change.

**Comment by simonjayhawkins** [comment]
@jreback ok to merge (and backport)

**Review by simonjayhawkins — APPROVED** [review]

File: pandas/core/arrays/sparse/array.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ class SparseArray(PandasObject, ExtensionArray, ExtensionOpsMixin):
         3. ``data.dtype.fill_value`` if `fill_value` is None and `dtype`
            is not a ``SparseDtype`` and `data` is a ``SparseArray``.
 
-    kind : {'int', 'block'}, default 'int'
+    kind : {'integer', 'block'}, default 'integer'
```
Comment (jreback) [inline pandas/core/arrays/sparse/array.py:237]: is this actually true?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/sparse/array.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/arrays/sparse/array.py:237]: the default is 'integer'

https://github.com/pandas-dev/pandas/blob/fe1ea14910c9afec638114fce12308cdf26784dd/pandas/core/arrays/sparse/array.py#L277-L286

and is passed onto `make_sparse`

https://github.com/pandas-dev/pandas/blob/fe1ea14910c9afec638114fce12308cdf26784dd/pandas/core/arrays/sparse/array.py#L1513-L1528

**Review by jreback — APPROVED** [review]

**Comment by simonjayhawkins** [comment]
Thanks @edwardkong

**Pull request merged**